test/modules/layers/core-layers.spec.js

```diff
@@ -34,336 +33,123 @@ import {
 
 import * as FIXTURES from 'deck.gl/test/data';
 
-const getPointPosition = d => d.COORDINATES;
+import {autoTestLayer} from './test-utils';
 
-test('ScreenGridLayer#constructor', t => {
-  const data = FIXTURES.points;
-
-  testLayer({
+test('ScreenGridLayer', t => {
+  autoTestLayer(t, {
```

Comment:
I would much prefer it if we could keep the `testLayer` and just have a function that autogenerates test cases:

```
testLayer({
  Layer: ScreenGridLayer,
  testCases: [
    ...autoGenerateTestCases(ScreenGridLayer),
    {
       ...
```

Every time I see a new function layered on instead of the documented API, I need to try to figure out what is going in the wrapping function, if it is just some thing trivial or if there is something more to it. In my opinion, by constraining the function as proposed, it become much clearer what is happening:

Reply:
Properly named, this function could perhaps even be part of test utils module?